go-controller/pkg/node/management-port-dpu.go

```diff
@@ -165,20 +175,34 @@ func newManagementPortNetdev(hostSubnets []*net.IPNet) ManagementPort {
 }
 
 func (mp *managementPortNetdev) Create(nodeAnnotator kube.Annotator, waiter *startupWaiter) (*managementPortConfig, error) {
-	// get Netdev that is used for management port.
+	klog.Infof("Lookup netdevice link and existing management port")
+	// get netdev that is used for management port.
 	link, err := util.GetNetLinkOps().LinkByName(mp.netdevName)
 	if err != nil {
+		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
+			return nil, fmt.Errorf("failed to lookup %s link: %v", mp.netdevName, err)
+		}
 		// this may not the first time invoked on the node after reboot
 		// netdev may have already been renamed to ovn-k8s-mp0.
 		link, err = util.GetNetLinkOps().LinkByName(types.K8sMgmtIntfName)
 		if err != nil {
 			return nil, fmt.Errorf("failed to get link device for %s. %v", mp.netdevName, err)
 		}
+	} else if mp.netdevName != types.K8sMgmtIntfName {
+		if config.OvnKubeNode.Mode == types.NodeModeDPUHost {
+			// We do not expect OVS running here so just check existing interface
+			err = unconfigurePlatformMgmtInterface(mp.hostSubnets, types.K8sMgmtIntfName)
```

Comment:
the comment says check existing interface, but then you configure the mgmt port. I'm confused.

Reply:
I'll change the comment then